controllers/akodeploymentconfig/cluster/cluster_controller_addon_secret.go

```diff
@@ -114,6 +114,18 @@ func AkoAddonSecretDataYaml(cluster *clusterv1.Cluster, obj *akoov1alpha1.AKODep
 	if err != nil {
 		return "", err
 	}
+
+	//Avoid setting DeleteConfig for management cluster
+	if cluster.Namespace != akoov1alpha1.TKGSystemNamespace {
+		if deleteConfig, exists := cluster.Labels[akoov1alpha1.AviClusterDeleteConfigLabel]; exists {
```

Comment:
who will be responsible for setting the `AviClusterDeleteConfigLabel` for cluster? during migration?

Reply:
user/migration script